spandecorator/handler.go

```diff
@@ -0,0 +1,79 @@
+package spandecorator
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/SKF/go-utility/v2/log"
+	"github.com/SKF/go-utility/v2/useridcontext"
+
+	middleware "github.com/SKF/go-enlight-middleware"
+)
+
+const UserIDKey = "userId"
+
+type Middleware struct {
+	Tracer middleware.Tracer
+}
+
+func New() *Middleware {
+	return &Middleware{Tracer: &middleware.OpenCensusTracer{}}
+}
+
+func (m *Middleware) Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			span := m.Tracer.SpanFromContext(r.Context())
+			attrs := getAttributes(r)
+			for k, v := range attrs {
+				span.AddStringAttribute(k, v)
+			}
+
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func getAttributes(r *http.Request) map[string]string {
```

Comment:
Maybe make this a method on the middleware?

Might be nice to take in the context here instead of using the one from the request

Reply:
Why? I am not using any state here?

Remove the use of context. Since it is not an error to be unable to extract userID anymore.

Reply:
I was thinking for some future improvements, but it can probably be done in the same PR with the improvements since we don't export it.